cookbooks/aws-parallelcluster-slurm/recipes/update_shared_storages.rb

```diff
@@ -0,0 +1,165 @@
+      not_in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+    end
+    shared_dir_array = []
+    vol_array = []
+    unless in_shared_storages_mapping["ebs"].nil?
+      in_shared_storages_mapping["ebs"].each do |storage|
+        next unless not_in_shared_storages_mapping["ebs"].nil? || !not_in_shared_storages_mapping["ebs"].include?(storage)
+        shared_dir_array.push(storage["mount_dir"])
+        vol_array.push(storage["volume_id"])
+      end
+    end
+    [shared_dir_array, vol_array]
+  end
+
+  def get_efs(action)
+    # get efs to mount or unmount
+    if action == "unmount"
+      in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+    elsif action == "mount"
+      in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+    end
+    shared_dir_array = []
+    efs_fs_id_array = []
+    unless in_shared_storages_mapping["efs"].nil?
+      in_shared_storages_mapping["efs"].each do |storage|
+        next unless not_in_shared_storages_mapping["efs"].nil? || !not_in_shared_storages_mapping["efs"].include?(storage)
+        shared_dir_array.push(storage["mount_dir"])
+        efs_fs_id_array.push(storage["efs_fs_id"])
+      end
+    end
+    [shared_dir_array, efs_fs_id_array]
+  end
+
+  def get_fsx(action)
+    # get fsx to mount or unmount
+    if action == "unmount"
+      in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+    elsif action == "mount"
+      in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+    end
+    fsx_fs_id_array = []
+    fsx_fs_type_array = []
+    fsx_shared_dir_array = []
+    fsx_dns_name_array = []
+    fsx_mount_name_array = []
+    fsx_volume_junction_path_array = []
+    unless in_shared_storages_mapping["fsx"].nil?
+      in_shared_storages_mapping["fsx"].each do |storage|
+        next unless not_in_shared_storages_mapping["fsx"].nil? || !not_in_shared_storages_mapping["fsx"].include?(storage)
+
+        fsx_fs_id_array.push(storage["fsx_fs_id"])
+        fsx_fs_type_array.push(storage["fsx_fs_type"])
+        fsx_shared_dir_array.push(storage["mount_dir"])
+        fsx_dns_name_array.push(storage["fsx_dns_name"])
+        fsx_mount_name_array.push(storage["fsx_mount_name"])
+        fsx_volume_junction_path_array.push(storage["fsx_volume_junction_path"])
+      end
+    end
+    [fsx_fs_id_array, fsx_fs_type_array, fsx_shared_dir_array, fsx_dns_name_array, fsx_mount_name_array, fsx_volume_junction_path_array]
+  end
+end
+# unmount ebs
+manage_ebs "ebs" do
```

Comment:
Here and below: what about having a more explanatory block descriptions, e.g. "Unmount EBS volume #{volume_id}" instead of "ebs"?

Reply:
Change the description to `Remove EBS`, `Add EBS`